pkg/scalers/datadog_scaler.go

```diff
@@ -195,66 +222,68 @@ func (s *datadogScaler) IsActive(ctx context.Context) (bool, error) {
 			"site": s.metadata.datadogSite,
 		})
 
-	resp, _, err := s.apiClient.MetricsApi.QueryMetrics(ctx, time.Now().Unix()-int64(s.metadata.age), time.Now().Unix(), s.metadata.query)
+	resp, r, err := s.apiClient.MetricsApi.QueryMetrics(ctx, time.Now().Unix()-int64(s.metadata.age), time.Now().Unix(), s.metadata.query)
+
+	if r.StatusCode == 429 {
+		rateLimit := r.Header.Get("X-Ratelimit-Limit")
+		rateLimitReset := r.Header.Get("X-Ratelimit-Reset")
+
+		return -1, fmt.Errorf("your Datadog account reached the %s queries per hour rate limit, next limit reset will happen in %s seconds: %s", rateLimit, rateLimitReset, err)
+	}
+
+	if r.StatusCode != 200 {
+		return -1, fmt.Errorf("error when retrieving Datadog metrics: %s", err)
+	}
 
 	if err != nil {
-		return false, err
+		return -1, fmt.Errorf("error when retrieving Datadog metrics: %s", err)
 	}
 
 	series := resp.GetSeries()
+	tries := 0
 
-	if len(series) == 0 {
-		return false, nil
-	}
+	// Sometimes Datadog won't be returning a series for a particular call, if the data is not there yet, let's retry 5 times before giving up
+	for len(series) == 0 && tries < 5 {
+		time.Sleep(5 * time.Second)
 
-	points := series[0].GetPointlist()
+		resp, r, err := s.apiClient.MetricsApi.QueryMetrics(ctx, time.Now().Unix()-int64(s.metadata.age), time.Now().Unix(), s.metadata.query)
 
-	if len(points) == 0 {
-		return false, nil
-	}
+		if r.StatusCode == 429 {
+			rateLimit := r.Header.Get("X-Ratelimit-Limit")
+			rateLimitReset := r.Header.Get("X-Ratelimit-Reset")
 
-	return true, nil
-}
+			return -1, fmt.Errorf("your Datadog account reached the %s queries per hour rate limit, next limit reset will happen in %s seconds: %s", rateLimit, rateLimitReset, err)
+		}
```

Comment:
I wouldn't do this. HPA controller runs in single thread and with this we are blocking it for up to 25 seconds. This could produce impact in the whole cluster

Reply:
Yeah, the call should be definitely non blocking.